third_party/move/move-compiler-v2/src/pipeline/reference_safety_processor.rs

```diff
@@ -1806,26 +1827,11 @@ impl<'env> TransferFunctions for LifeTimeAnalysis<'env> {
         // Construct step context
         let mut step = self.new_step(code_offset, instr.get_attr_id(), state);
 
-        // Preprocessing: release all temps in the label map which are no longer alive at this point.
-        step.state.debug_print("before enter release");
-        let alive_temps = step.alive.before_set();
-        for temp in step
-            .state
-            .temp_to_label_map
-            .keys()
-            .cloned()
-            .collect::<Vec<_>>()
-        {
-            if !alive_temps.contains(&temp) && step.is_ref(temp) {
-                step.state.release_ref(temp)
-            }
-        }
-
-        // Preprocessing: check borrow safety of the currently active borrow graph for read ref,
-        // write ref, and function calls.
+        // Preprocessing: check borrow safety of the currently active borrow graph for
```

Comment:
Since release_refs_not_alive_after is called after handling this instr, this is not needed anymore?

Reply:
It was always like a defensive meaure 'if something has been not release by now, do it now', but it should not be required as releasing happens (before and after) at the end of each instruction, with or without this new function. I removed it and verified everything works like before. 

The difference is really that the cleanup at the end and what you did with your 'release before write' is now unified in `release_refs_not_alive_after`. The special check whether the write destination is not a ref is also not longer needed, by doing the releasing after the output graph has been created.